backend/Functions/Edna.AssignmentLinks/Edna.AssignmentLinks/AssignmentLinksApi.cs

```diff
@@ -103,18 +125,81 @@ public AssignmentLinksApi(ILogger<AssignmentLinksApi> logger, IMapper mapper)
         public async Task<IActionResult> DeleteLink(
             [HttpTrigger(AuthorizationLevel.Anonymous, "delete", Route = "assignments/{assignmentId}/links/{linkId}")] HttpRequest req,
             [Table(AssignmentLinksTableName)] CloudTable assignmentLinksTable,
-            [Table(AssignmentLinksTableName, "{assignmentId}", "{linkId}")] AssignmentLinkEntity entityToDelete)
+            [Table(AssignmentLinksTableName, "{assignmentId}", "{linkId}")] AssignmentLinkEntity entityToDelete,
+            [Assignment(AssignmentId = "{assignmentId}")] Assignment assignment,
+            [Platform] PlatformsClient platformsClient,
+            [LtiAdvantage] INrpsClient nrpsClient,
+            [Lti1] Lti1MembershipClient membershipClient)
         {
             if (entityToDelete == null)
+            {
+                _logger.LogInformation("Entity to delete is null");
                 return new OkResult();
+            }
+
+            IActionResult res = await ValidateUser(req, assignment, platformsClient, nrpsClient, membershipClient);
+            if (res.GetType() != typeof(OkResult))
+                return res;
 
             TableOperation deleteTable = TableOperation.Delete(entityToDelete);
             TableResult deleteResult = await assignmentLinksTable.ExecuteAsync(deleteTable);
 
             if (deleteResult.HttpStatusCode < 200 || deleteResult.HttpStatusCode >= 300)
                 return new InternalServerErrorResult();
 
+            _logger.LogInformation("Link deleted.");
+
+            return new OkResult();
+        }
+
+        private async Task<IActionResult> ValidateUser(HttpRequest req, Assignment assignment, PlatformsClient platformsClient, INrpsClient nrpsClient, Lti1MembershipClient membershipClient)
+        {
+            if (!req.Headers.TryGetUserEmails(out List<string> userEmails))

```

Comment:
🚫 The name `TryGetUserEmails` seems a bit confusing and totally shadows the System Identity use-case (i.e. `{true, []}` case). If I think a bit more, it is essentially hiding the fact that we are using it to understand the type of caller we are dealing with. 
> While I'd think harder to refactor this function signature to be more expressive, please think a bit on your end as well.

Reply:
I don't remember my thought process around this, but I do believe it is something related to the function doing something more than just returning the `UserEmails` -- like also checking whether the api call made from another system or a client based on `appidacr`. If yes, can we try to come up with a better suited name?

Reply:
Yes, if I remember correctly, this was the reason.
Out of the top of my head, we could rename it as `ValidateRequest`

Reply:
The recommended way should be to have 2 separate calls, one for `GetClientAuthenticationType` (public, clientID or Certificate type) and the other being `TryGetUserEmails`. This way it would be much clear about our intent and the function would have single responsibility as well. Given that the change in itself is an entity, it would be fine to take it separately. However, may I request to have it taken up immediately after we merge this?